ord_schema/updates.py

```diff
@@ -0,0 +1,128 @@
+                    smiles = _pubchem_resolve('name', identifier.value)
+                    new_identifier = compound.identifiers.add()
+                    new_identifier.type = new_identifier.SMILES
+                    new_identifier.value = smiles
+                    new_identifier.details = 'NAME resolved by PubChem'
+                    modified = True
+                    break
+                except urllib.error.HTTPError:
+                    pass
+    return modified
+
+
+def add_binary_identifiers(message):
+    """Adds RDKIT_BINARY identifiers for compounds with valid structures.
+
+    Note that the RDKIT_BINARY representations are mostly useful in the context
+    of searching the database. Accordingly, this function is not included in the
+    standard set of Reaction updates in update_reaction().
+
+    Args:
+        message: Reaction proto.
+
+    Returns:
+        Boolean whether `message` was modified.
+    """
+    modified = False
+    compounds = message_helpers.find_submessages(message, reaction_pb2.Compound)
+    for compound in compounds:
+        if any(identifier.type == identifier.RDKIT_BINARY
+               for identifier in message.identifiers):
+            continue
+        for identifier in compound.identifiers:
+            mol = None
+            if Chem and identifier.type == identifier.SMILES:
+                mol = Chem.MolFromSmiles(identifier.value)
+            elif identifier.type == identifier.INCHI:
+                mol = Chem.MolFromInchi(identifier.value)
+            elif identifier.type == identifier.MOLBLOCK:
+                mol = Chem.MolFromMolBlock(identifier.value)
+            if mol is not None:
+                compound.identifiers.add(
+                    bytes_value=mol.ToBinary(), type='RDKIT_BINARY')
+                modified = True
+    return modified
+
+
+def update_reaction(reaction):
+    """Updates a Reaction message.
+
+    Current updates:
+      * Sets record_id if not already set.
+
+    Args:
+        reaction: reaction_pb2.Reaction message.
+    """
+    # NOTE(kearnes): Provenance updates do not trigger "modified" events.
+    if not reaction.provenance.HasField('record_created'):
+        reaction.provenance.record_created.time.value = (
+            datetime.datetime.utcnow().ctime())
+    # TODO(kearnes): There's more complexity here regarding record_id values
+    # that are set outside of the submission pipeline. It's not as simple as
+    # requiring them to be empty, since a submission may include edits to
+    # existing reactions.
+    if not reaction.provenance.record_id:
+        record_id = f'ord-{uuid.uuid4().hex}'
+        reaction.provenance.record_id = record_id
+    modified = False
+    for func in _UPDATES:
+        modified = modified or func(reaction)
+    if modified:
+        reaction.provenance.record_modified.add().time.value = (
+            datetime.datetime.utcnow().ctime())
+
+
+# Standard updates.
+_UPDATES = [
+    resolve_names,
```

Comment:
`add_binary_identifiers` too?

Reply:
I was thinking that maybe we'd leave this out and only put it in for uploading to BigQuery, etc. It adds a fair amount of noise to the submission to have commits that add a bunch of unreadable bytes and I don't think there's much benefit to having the RDKIT_BINARY reps in typical offline ML settings. (There's a comment about this in the docstring for add_binary_identifiers.)

What do you think?

Reply:
I actually like the idea of `RDKIT_BINARY` being present, since it makes it very convenient to use the data for ML in Python if you reload the compounds from binary (and are using RDKit, of course). 

This might not matter as much if we are primarily working with SMILES and MolBlocks. I also understand the drawback of having unreadable bytes. But having a pre-parsed RDKit object is quite convenient

Reply:
I'll suggest adding this post-review in a separate PR if we want it as part of the official record. That way we don't need to include it as part of the submission review.